### PR TITLE
VTS L1 update to Sepc 1.3.0

### PR DESCRIPTION
**Pull request by akhilbhas**
Updating TV Settings L1 cases according to the spec update in 1.3.0

### CODE REVIEW

File: src/test_l1_tvSettings.c [hunk 1]
```diff
@@ -12701,9 +12701,10 @@ void test_l1_tvSettings_negative_SaveDvTmaxValue (void)
...
  * | :-------: | ------------- | --------- | --------------- | ----- |
  * | 01 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
- * | 02 | call GetSupportedComponentColor() -  Retrieve the current Supported ComponentColor and validate Supported ComponentColor by looping through the ComponentColor section of test specific config file | int * | (tvERROR_NONE | tvERROR_OPERATION_NOT_SUPPORTED) | Should Pass |
- * | 03 | call GetSupportedComponentColor() -  Retrieve the current Supported ComponentColor with valid argument and validate with above value | int * | tvERROR_NONE | Should Pass |
- * | 04 | call TvTerm() -  Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
+ * | 02 | call SetCMSState() - Set the CMSState with valid value | true | tvERROR_NONE | Should Pass |
+ * | 03 | call GetSupportedComponentColor() -  Retrieve the current Supported ComponentColor and validate Supported ComponentColor by looping through the ComponentColor section of test specific config file | int * | (tvERROR_NONE | tvERROR_OPERATION_NOT_SUPPORTED) | Should Pass |
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:12705]: test_l1_tvSettings_positive_GetCurrentVideoResolution  - GetCurrentVideoResolution -The API returns the default value as tvVideoResolution_NONE, which needs to be validated. In interface document it is stated as the default output parameter.

Same way

test_l1_tvSettings_positive_GetCurrentVideoFrameRate  - GetCurrentVideoFrameRate - default out param is tvVideoFrameRate_NONE

test_l1_tvSettings_positive_GetCurrentVideoSource - GetCurrentVideoSource - default out param is VIDEO_SOURCE_IP

test_l1_tvSettings_positive_GetCurrentVideoFormat  - GetCurrentVideoFormat - default out param is VIDEO_FORMAT_SDR

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_tvSettings.c (hunk 1 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c:12673]: Updated.

File: src/test_l1_tvSettings.c [hunk 2]
```diff
@@ -12716,9 +12685,13 @@ void test_l1_tvSettings_positive_GetSupportedComponentColor (void)
 
 	/* Step 01: Calling tvsettings initialization and expecting the API to return success */
 	result = TvInit();
-	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
+	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);	
+
+	/* Step 02: Calling tvsettings to Set the SetCMSState for value true and the API to return success */
+	result = SetCMSState(true);
+	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_NONE);
```
Comment (thanushreevani) [inline src/test_l1_tvSettings.c:12692]: I think this SaveCMSState(true) has to be for SetCurrentComponentSaturation and not for GetCurrentComponentSaturation

File: src/test_l1_tvSettings.c [hunk 3]
```diff
@@ -12980,11 +12954,15 @@ void test_l1_tvSettings_positive_GetCurrentComponentSaturation (void)
 	result = TvInit();
 	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
 
-	/* Step 02: Calling tvsettings GetCurrentComponentSaturation and expecting the API to return success */
-	result = GetCurrentComponentSaturation((tvDataComponentColor_t)Configfile.componentColor.modeId[0],&saturation);
+	/* Step 02: Calling tvsettings to Set the SetCMSState for value true and the API to return success */
+	result = SetCMSState(true);
```
Comment (thanushreevani) [inline src/test_l1_tvSettings.c:12958]: I think this SaveCMSState(true) has to be for SetCurrentComponentSaturation and not for GetCurrentComponentSaturation

File: src/test_l1_tvSettings.c [hunk 4]
```diff
@@ -13271,11 +13250,15 @@ void test_l1_tvSettings_positive_GetCurrentComponentHue (void)
 	result = TvInit();
 	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
 
-	/* Step 02: Calling tvsettings GetCurrentComponentHue and expecting the API to return success */
-	result = GetCurrentComponentHue((tvDataComponentColor_t)Configfile.componentColor.modeId[0],&Hue);
+	/* Step 02: Calling tvsettings to Set the SetCMSState for value true and the API to return success */
+	result = SetCMSState(true);
```
Comment (thanushreevani) [inline src/test_l1_tvSettings.c:13254]: same for Hue too

File: src/test_l1_tvSettings.c [hunk 5]
```diff
@@ -13288,7 +13271,7 @@ void test_l1_tvSettings_positive_GetCurrentComponentHue (void)
 		UT_ASSERT_AUTO_TERM_TRUE( (Hue >=0 && Hue <= 100));
 	}
 
-	/* Step 04: Calling tvsettings termination and expecting the API to return success */
+	/* Step 05: Calling tvsettings termination and expecting the API to return success */
 	result = TvTerm();
```
Comment (thanushreevani) [inline src/test_l1_tvSettings.c:13275]: similarly add for SetCMSState(true) for SetCurrentComponentLuma

File: src/test_l1_tvSettings.c (hunk 2 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c]: Moved to SetCurrentComponentSaturation

File: src/test_l1_tvSettings.c (hunk 3 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c]: moved to setCurrentComponentHue

File: src/test_l1_tvSettings.c [hunk 6]
```diff
@@ -12716,9 +12684,10 @@ void test_l1_tvSettings_positive_GetSupportedComponentColor (void)
 
 	/* Step 01: Calling tvsettings initialization and expecting the API to return success */
 	result = TvInit();
-	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
+	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);	
+
 
-	/* Step 02: Calling tvsettings GetSupportedComponentColor and expectinging the API to return success */
+	/* Step 03: Calling tvsettings GetSupportedComponentColor and expectinging the API to return success */
```
Comment (thanushreevani) [inline src/test_l1_tvSettings.c:12690]: updating the steps number can be reverted

File: src/test_l1_tvSettings.c [hunk 7]
```diff
@@ -12980,11 +12954,11 @@ void test_l1_tvSettings_positive_GetCurrentComponentSaturation (void)
 	result = TvInit();
 	UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
 
-	/* Step 02: Calling tvsettings GetCurrentComponentSaturation and expecting the API to return success */
+	/* Step 03: Calling tvsettings GetCurrentComponentSaturation and expecting the API to return success */
 	result = GetCurrentComponentSaturation((tvDataComponentColor_t)Configfile.componentColor.modeId[0],&saturation);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_NONE);
 
-	/* Step 03: Calling tvsettings GetCurrentComponentSaturation and expecting the API to return success */
+	/* Step 04: Calling tvsettings GetCurrentComponentSaturation and expecting the API to return success */
```
Comment (thanushreevani) [inline src/test_l1_tvSettings.c]: updating the steps number can be reverted

File: src/test_l1_tvSettings.c (hunk 4 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c]: Moved to SetCurrentComponentSaturation

File: src/test_l1_tvSettings.c (hunk 5 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c]: Moved to SetCurrentComponentLuma

File: src/test_l1_tvSettings.c (hunk 6 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c]: reverted

File: src/test_l1_tvSettings.c (hunk 7 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c]: Reverted

File: src/test_l1_tvSettings.c [hunk 8]
```diff
@@ -1223,20 +1215,13 @@ void test_l1_tvSettings_positive_GetCurrentVideoResolution (void)
...
 
 	/* Step 03: Calling tvsettings GetCurrentVideoFormat and expectinging the API to return success */
 	result = GetCurrentVideoResolution(&tvVideoResolutionRetry);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_NONE);
 
-	UT_ASSERT_AUTO_TERM_NUMERICAL(tvVideoResolutionRetry.resolutionValue, tvVideoResolution.resolutionValue);
-	UT_ASSERT_AUTO_TERM_NUMERICAL(tvVideoResolutionRetry.frameHeight, tvVideoResolution.frameHeight);
-	UT_ASSERT_AUTO_TERM_NUMERICAL(tvVideoResolutionRetry.frameWidth, tvVideoResolution.frameWidth);
-	UT_ASSERT_AUTO_TERM_NUMERICAL(tvVideoResolutionRetry.isInterlaced, tvVideoResolution.isInterlaced);
+	UT_ASSERT_EQUAL(tvVideoResolutionRetry.resolutionValue, tvVideoResolution.resolutionValue);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:1224]: test_l1_tvSettings_positive_GetCurrentVideoFrameRate  - Please modify the step 2 in the description as we are now reading the default out parameter. 

| 02 | call GetCurrentVideoFrameRate() -  Retrieve the current VideoFrameRate and validate VideoFrameRate by looping through the VideoFramerate section of test specific config file values

Similar way, please check for the other test functions description.

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_tvSettings.c (hunk 8 above)
Comment (akhilbhas) [reply src/test_l1_tvSettings.c:1222]: updated.

**Review by thanushreevani — APPROVED** [review]

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Pull request merged**